src/app/pages/PerpetualPage/components/TradeReviewDialog/index.tsx

```diff
@@ -39,15 +71,188 @@ export const TradeReviewDialog: React.FC = () => {
     [openTrade, trade],
   );
 
+  const margin = useMemo(() => {
+    if (!trade) {
+      return 0;
+    }
+
+    const amount =
+      Number(fromWei(trade.amount)) * getTradeDirection(trade.position);
+
+    return getRequiredMarginCollateral(
+      trade!.leverage,
```

Comment:
Shouldn't be needed here.

```suggestion
      trade.leverage,
```